extension/bearertokenextension/config.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bearertokenextension
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+// Config specifies how the Per-RPC bearer token based authentication data should be obtained.
+type Config struct {
+	config.ExtensionSettings `mapstructure:",squash"`
+
+	// BearerToken specifies the bearer token to use for every RPC.
+	BearerToken string `mapstructure:"bearer_token,omitempty"`
```

Comment:
Given that the name of the extension will include the word "bearer", should this field just be called "token" so that it's not repetitive?

Reply:
+1